x-pack/legacy/plugins/reporting/server/plugin.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Legacy } from 'kibana';
+import { CoreSetup, CoreStart, Plugin } from 'src/core/server';
+import { IUiSettingsClient } from 'src/core/server';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { XPackMainPlugin } from '../../xpack_main/server/xpack_main';
+// @ts-ignore
+import { mirrorPluginStatus } from '../../../server/lib/mirror_plugin_status';
+import { PLUGIN_ID } from '../common/constants';
+import { ReportingPluginSpecOptions } from '../types.d';
+import { registerRoutes } from './routes';
+import { LevelLogger, checkLicenseFactory, getExportTypesRegistry, runValidations } from './lib';
+import { createBrowserDriverFactory } from './browsers';
+import { registerReportingUsageCollector } from './usage';
+import { logConfiguration } from '../log_configuration';
+
+// For now there is no exposed functionality to other plugins
```

Comment:
In the near future, we will expose functionality to other plugins so they have an API to schedule reports of their data visualizations / CSV exports